### PR TITLE
Issue #2659 - Fix 1.3 login regressions

### PR DESCRIPTION
**Pull request by alexmt**
Checklist:

* [x]  (b) this is a bug fix
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Optional. My organization is added to the README.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)). 

#2659

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -492,16 +492,17 @@ func (a *ArgoCDServer) translateGrpcCookieHeader(ctx context.Context, w http.Res
 		}
 		token := sessionResp.Token
 		if token != "" {
-			token, err := zjwt.ZJWT(token)
```
Comment (alexec) [inline server/server.go]: tests?

File: ui/src/app/shared/components/page.tsx [hunk 2]
```diff
@@ -21,16 +20,19 @@ export class Page extends React.Component<{title: string; toolbar?: Toolbar | Ob
...
-                            parse(document.cookie)['argocd.token'] ? (
-                                <a key='logout' onClick={() => this.goToLogin(true)}>
-                                    Logout
-                                </a>
-                            ) : (
-                                <a key='login' onClick={() => this.goToLogin(false)}>
-                                    Login
-                                </a>
-                            )
+                            <DataLoader key='loginPanel' load={() => services.users.get()}>
```
Comment (alexec) [inline ui/src/app/shared/components/page.tsx]: nice

File: ui/src/app/app.tsx [hunk 3]
```diff
@@ -67,18 +67,25 @@ async function isExpiredSSO() {
...
-            if (await isExpiredSSO()) {
-                window.location.href = `${basehref}/auth/login?return_url=${encodeURIComponent(location.href)}`;
-            } else {
-                history.push(`${basehref}/login?return_url=${encodeURIComponent(location.href)}`);
-            }
+        if (history.location.pathname.startsWith('/login')) {
+            return;
+        }
+
+        const isSSO = await isExpiredSSO();
```
Comment (alexec) [inline ui/src/app/app.tsx]: tests?

File: server/server.go (hunk 1 above)
Comment (alexmt) [reply server/server.go]: added

File: ui/src/app/app.tsx (hunk 3 above)
Comment (alexmt) [reply ui/src/app/app.tsx]: I think this is only possible to test is in e2e test ( Even if we  could test app.tsx ) . Will do it once we get to UI e2e tests

Comment (alexec) [reply ui/src/app/app.tsx]: if you approve #2610  you can do this without maybe?

**Review by alexec — APPROVED** [review]

**Pull request merged**